Release ironwood

slandrault opened the pull request:
## Purpose

This PR intends to update the edxapp to the first release of ironwood (2-oee).
It also update a script to not use a dump of the base's schema. This was done before to save time as migrate were too long. Migrations have been squashed in ironwood so it should be faster now.

slandrault committed:
⬆️(edxapp) bump edxapp to ironwood.2-oee-1.0.1

Upgrade edxapp to current major release, ironwood-2.

slandrault committed:
🔖(minor) bump release to 5.15.0

Bump release to 5.15.0

- Upgrade `ansible` to `2.9.13`
- Upgrade `edxapp` version to `ironwood.2-1.0.1`

jmaupetit commented on CHANGELOG.md:
```diff
@@ -8,9 +8,16 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 ### Changed
 
 - Upgrade `ansible` to `2.9.13`
+- Upgrade `edxapp` version to `ironwood.2-1.0.1`
```
```suggestion
- Upgrade default `edxapp` version to `ironwood.2-1.0.1`
```

jmaupetit commented on group_vars/all/main.yml:
```diff
@@ -46,6 +46,12 @@ trashable_env_types:
+# This is useful when migration become too long to execute. We can
+# provide a sql dump of the schema to then load it instead of waiting
+# for migrate to be done (works only with trashable env)
+enable_sql_dump_loading: false
```
If it defaults to `false` you are introducing a BC.

slandrault replied:
Right good point

jmaupetit commented on group_vars/all/main.yml:
```diff
@@ -46,6 +46,12 @@ trashable_env_types:
   - "feature"
 
+# To be use with edxapp_sql_dump_url in edxapp config
+# This is useful when migration become too long to execute. We can
```
```suggestion
# This is useful when migrations become too long to execute. We can
```

jmaupetit commented on group_vars/all/main.yml:
```diff
@@ -46,6 +46,12 @@ trashable_env_types:
+# To be use with edxapp_sql_dump_url in edxapp config
+# This is useful when migration become too long to execute. We can
+# provide a sql dump of the schema to then load it instead of waiting
+# for migrate to be done (works only with trashable env)
```
```suggestion
# for the migrate command to end (works only for trashable environments)
```

jmaupetit commented on CHANGELOG.md:
```diff
@@ -8,9 +8,16 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
 
+## [5.14.3] - 2020-09-17
```
This is not a fix, so you need to release a minor version:

```suggestion
## [5.15.0] - 2020-09-17
```

lunika commented on CHANGELOG.md:
```diff
@@ -8,6 +8,10 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
```
In your commit message:

`s/To be use with/To be used with/`

`s/when migration become too long/when the migration becomes too long/`

`s/We can provide a sql dump of the schema to then load it/We can provide a sql dump of the schema and then load it`

lunika replied:
In the commit title, the subject is not `main` but `edxapp`. It is only related to edxapp application.

lunika commented on group_vars/all/main.yml:
```diff
@@ -46,6 +46,12 @@ trashable_env_types:
   - "development"
   - "feature"
 
+# To be use with edxapp_sql_dump_url in edxapp config
```
```suggestion
# To be used with edxapp_sql_dump_url in edxapp config
```

lunika commented on group_vars/all/main.yml:
```diff
@@ -46,6 +46,12 @@ trashable_env_types:
 
+# To be use with edxapp_sql_dump_url in edxapp config
+# This is useful when migrations become too long to execute. We can
+# provide a sql dump of the schema to then load it instead of waiting
```
```suggestion
# provide a sql dump of the schema and then load it instead of waiting
```

lunika commented on CHANGELOG.md:
```diff
@@ -8,14 +8,16 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
```
Your commit title does not respect our release commits, should be:

`🔖(minor) bump release to 5.15.0`

jmaupetit commented on group_vars/customer/eugene/development/secrets/kibana.vault.yml:
```diff
@@ -0,0 +1,21 @@
+32323364373936316138326564356663356466366434623231663663663964653431313638356464
+66653864303363366433613038393337386133373065333335613039383030653035343830373065
+38323164383964623136643635613430316239363766323232623533636139383562626430376166
+38623437396366643632
```
Why did you add this vault to this commit? Sounds like it comes out of the blue :sweat_smile:

slandrault committed:
✅(edxapp) fix lms test to be compliant with ironwood and hawthorn

When upgrading to ironwood, the lms page welcome text has changed.
This commit will reduce the tested message from
"It works! Welcome..." to simply "It works!" (while still checking
for a 200 status code).

jmaupetit reviewed (changes requested):
Almost good to go.

1. this is not a patch but a minor release (see unresolved comment)
2. In your commits:
  - 2358f50f515d19cedc285471fdba6ebd3d2e0993: in the message `s/next/current/`
  - 2fb4dc8003c03355459b24152cdfcef4d075b527:
    * the title emoji does not comply with a new release (see @lunika's comment)
    * the message content does not reflect changes from the CHANGELOG

jmaupetit commented on CHANGELOG.md:
```diff
@@ -10,6 +10,7 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 
 ### Changed
 
+- Update ci test for edxapp
```
As it's CI-related, I am not sure it's worth to be mentioned in the CHANGELOG.

jmaupetit commented on CHANGELOG.md:
```diff
@@ -8,6 +8,8 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
 
+## [5.14.3] - 2020-09-17
```
You should also fix the date :sweat_smile:

jmaupetit commented on apps/edxapp/tray.yml:
```diff
@@ -1,3 +1,3 @@
 metadata:
   name: edxapp
-  version: hawthorn.1-3.2.0
+  version: ironwood.2-1.0.1
```
You commit title should be updated accordingly.

jmaupetit reviewed (approved):
![](https://media3.giphy.com/media/tGokBeqpHuSgE/giphy.gif)
